src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -266,6 +240,37 @@ namespace winrt::TerminalApp::implementation
         CATCH_LOG();
     }
 
+    winrt::fire_and_forget TerminalPage::NewTerminalByDrop(winrt::Windows::UI::Xaml::DragEventArgs& e)
+    {
+        Windows::Foundation::Collections::IVectorView<Windows::Storage::IStorageItem> items;
+        try
+        {
+            items = co_await e.DataView().GetStorageItemsAsync();
+        }
+        CATCH_LOG();
+
+        if (items.Size() == 1)
+        {
+            std::filesystem::path path(items.GetAt(0).Path().c_str());
+            if (!std::filesystem::is_directory(path))
```

Comment:
Wonder how well `std::filesystem::is_directory` handles UNC paths.

Reply:
tested it with `\\HOST\folder` and `\\HOST\folder\file.txt`, `std::filesystem::is_directory` handles it just fine.
Sadly `darg & drop` this UNC path only works with `windows powershell`.
Command Prompt gives the following error:
```
'\\HOST\folder'
CMD.EXE was started with the above path as the current directory.
UNC paths are not supported.  Defaulting to Windows directory.
Microsoft Windows [Version 10.0.19042.985]
(c) Microsoft Corporation. All rights reserved.

C:\Windows>
```
while `ubuntu`  just default to the home folder.